include/questdb/ingress/line_sender.hpp

```diff
@@ -333,6 +318,163 @@ namespace questdb::ingress
         int64_t _ts;
     };
 
+#if __cplusplus < 202002L
+    class buffer_view final
+    {
+      public:
+        using iterator = const std::byte*;
+        using const_iterator = const std::byte*;
+        using reverse_iterator = std::reverse_iterator<iterator>;
+        using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+        using reference = const std::byte&;
+
+      private:
+        const std::byte* buf{nullptr};
+        size_t len{0};
+
+      public:
+        // ------------------------------------------------------------------------
+        // Constructor & Core Accessors
+        // ------------------------------------------------------------------------
+
+        /// @brief Default constructor. Creates an empty buffer view.
+        buffer_view() noexcept = default;
+
+        /// @brief Constructs a buffer view from raw byte data.
+        /// @param data Pointer to the underlying byte array (may be nullptr if
+        /// length=0).
+        /// @param length Number of bytes in the array.
+        constexpr buffer_view(const std::byte* data, size_t length) noexcept
+            : buf(data), len(length)
+        {
+        }
+
+        /// @brief Gets a pointer to the underlying byte array.
+        /// @return Const pointer to the data (may be nullptr if empty()).
+        constexpr const std::byte* data() const noexcept { return buf; }
+
+        /// @brief Gets the number of bytes in the view.
+        /// @return Size in bytes.
+        constexpr size_t size() const noexcept { return len; }
+
+        /// @brief Checks if the view contains no bytes.
+        /// @return true if size() == 0.
+        constexpr bool empty() const noexcept { return len == 0; }
```

Comment:
Keep `data()`, `size()`, `empty()` and maybe the equality operator.

Remove the rest: It's too much.

We don't need to backport std-span, just provide a minimal API that's forward compatible with span.

The comments can also be cleaned up: Remove the section markers, too much as well.

Small is beautiful and easier to read :-)